graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+    name: 'LessonPlanField',
+    description: 'Fields that can be generated for a lesson plan',
+});
+
+@InputType()
+class GenerateLessonPlanInput {
+    @Field(() => [String], { nullable: true })
+    @ArrayMaxSize(MAX_FILES, { message: `Number of files must not exceed ${MAX_FILES}` })
+    fileUuids?: string[];
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => Int)
+    grade: number;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => pupil_state_enum)
+    state: pupil_state_enum;
+
+    @Field(() => String)
+    @MaxLength(MAX_PROMPT_LENGTH, { message: `Prompt must not exceed ${MAX_PROMPT_LENGTH} characters` })
+    prompt: string;
+
+    @Field(() => [LessonPlanField], { nullable: true })
+    expectedOutputs?: LessonPlanField[];
+
+    @Field(() => school_schooltype_enum)
+    schoolType: school_schooltype_enum;
+
+    @Field(() => String, { nullable: true, defaultValue: 'German' })
+    language?: string;
+}
+
+@ObjectType()
+class LessonPlanOutput {
+    @Field(() => String, { nullable: true })
+    title?: string;
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => String)
+    grade: string;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => String, { nullable: true })
+    learningGoal?: string;
+
+    @Field(() => String, { nullable: true })
+    agendaExercises?: string;
+
+    @Field(() => String, { nullable: true })
+    assessment?: string;
+
+    @Field(() => String, { nullable: true })
+    homework?: string;
+
+    @Field(() => String, { nullable: true })
+    resources?: string;
+}
+
+class LessonPlanGenerationError extends ApolloError {
+    constructor(message: string) {
+        super(message, 'LESSON_PLAN_GENERATION_ERROR');
+
+        Object.defineProperty(this, 'name', { value: 'LessonPlanGenerationError' });
+    }
+}
+
+// Define a type for the state codes based on the enum
+type StateCode = keyof typeof pupil_state_enum;
+
+// Create a mapping object for state codes to full names
+const stateMap: Record<StateCode, string> = {
+    bw: 'Baden-Württemberg',
+    by: 'Bayern',
+    be: 'Berlin',
+    bb: 'Brandenburg',
+    hb: 'Bremen',
+    hh: 'Hamburg',
+    he: 'Hessen',
+    mv: 'Mecklenburg-Vorpommern',
+    ni: 'Niedersachsen',
+    nw: 'Nordrhein-Westfalen',
+    rp: 'Rheinland-Pfalz',
+    sl: 'Saarland',
+    sn: 'Sachsen',
+    st: 'Sachsen-Anhalt',
+    sh: 'Schleswig-Holstein',
+    th: 'Thüringen',
+    at: 'Austria',
+    ch: 'Switzerland',
+    other: 'Other',
+};
+
+/**
+ * Gets the full name of a state from its code.
+ * @param stateCode - The state code.
+ * @returns The full name of the state.
+ */
+export function getStateFullName(stateCode: StateCode): string {
+    return stateMap[stateCode] || 'Unknown';
+}
+
+@Resolver()
+export class MutateLessonPlanResolver {
+    @Mutation(() => LessonPlanOutput)
+    @Authorized(Role.INSTRUCTOR, Role.ADMIN, Role.STUDENT)
```

Comment:
Role.STUDENT is not needed, all students involved in a match should have Role.INSTRUCTOR